oci/layout/oci_dest.go

```diff
@@ -184,23 +193,20 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 		Architecture: runtime.GOARCH,
 		OS:           runtime.GOOS,
 	}
-	d.index.Manifests = append(d.index.Manifests, desc)
+	d.addManifest(&desc)
 
 	return nil
 }
 
-func ensureDirectoryExists(path string) error {
-	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
-		if err := os.MkdirAll(path, 0755); err != nil {
-			return err
+func (d *ociImageDestination) addManifest(desc *imgspecv1.Descriptor){
+	for i, manifest := range d.index.Manifests {
+		if manifest.Digest.String() == desc.Digest.String() {
```

Comment:
The image name should not be part of the reference. The reference is the "tag". It doesn't make sense to include the path to the image, because you can move the image to a different path and because there's no cryptographic verification of image references.

Please, just use the current scheme. We had so many issues with references inside the OCI spec. I appreciate that the current spec doesn't forbid you from making a decision like that, I seriously would not recommend it.